src/controllers/newsController/news.js

```diff
@@ -1,18 +1,25 @@
+const client = require("../../cacheManager")
 const User = require("../../models/User/UserSchema")
 
 
 const getDefaultPreferencedNews = async(req,res)=>{
   console.log('getting default preferenced news' ,req.user)
   try{
-
     const preferences = req.user.news_preference
-    let promises = []
-    for(const preference of preferences){
-      promises.push(await fetch(`https://newsapi.org/v2/everything?q=${preference}&apiKey=${process.env.API_KEY}`))
-    }
-    // console.log(promises,' promises')
-    const finalData = await Promise.all(promises.map(promise=>promise.json()))
-    res.status(200).send({data: finalData})
+    let cachedPreferences = await client.get("user_preferences")
```

Comment:
user_preferences should be stored againt a user_id to identify and fetch key-value pairs quickly. We can have a composite key like 1234_user_preferences, where 1234 is the userid